src/structures/Webhook.js

```diff
@@ -235,18 +237,44 @@ class Webhook {
     return this;
   }
 
+  /**
+   * Options that can be passed into fetchMessage.
+   * @typedef {options} WebhookFetchMessageOptions
+   * @property {boolean} [cache=true] Whether to cache the message.
+   * @property {Snowflake} [threadId] The id of the thread this message belongs to.
+   * <info>For interaction webhooks, this property is ignored</info>
+   */
+
   /**
    * Gets a message that was sent by this webhook.
    * @param {Snowflake|'@original'} message The id of the message to fetch
-   * @param {boolean} [cache=true] Whether to cache the message
+   * @param {boolean|WebhookFetchMessageOptions} [cacheOrOptions=true|cacheOrOptions={}] If
+   * {@link ClientOptions#allowWebhookThreadFetching} is `true`, this will be {@link WebhookFetchMessageOptions}
+   * to enable fetching messages in threads. Otherwise, this will be the `cache` parameter.
    * @returns {Promise<Message|APIMessage>} Returns the raw message data if the webhook was instantiated as a
    * {@link WebhookClient} or if the channel is uncached, otherwise a {@link Message} will be returned
    */
-  async fetchMessage(message, cache = true) {
+  async fetchMessage(message, cacheOrOptions) {
```

Comment:
I really do not like this approach mainly because it’s hard to accurately display in typings and also unnecessarily complex. You should just add a third parameter and then change this in v14, maybe add a warning saying it will change

Reply:
You can't add a third option nicely here in a non-breaking way - it would have to come after `cache` which currently has a default, forcing them to always pass true/false in order to push to that third option. 

I think this is just best served by being a v14 PR either way.

Reply:
How about...keep this options, default it to `{ cache: true }`. Then instead of a client option, just accept the legacy option type by utilizing a `typeof options === 'object'`, and if its not let the old code run

Reply:
Or you could add the third parameter and pass undefined to the second one if you wanna use the default, I’m pretty sure that would work anyway

Reply:
That a horrible interface for the user.